generic_config_updater/generic_updater.py

```diff
@@ -0,0 +1,469 @@
+import json
+import jsonpatch
+import sonic_yang
+import os
+import copy
+from enum import Enum
+from swsssdk import ConfigDBConnector
+from imp import load_source
+load_source('sonic_cfggen', '/usr/local/bin/sonic-cfggen')
+from sonic_cfggen import deep_update, FormatConverter
+
+YANG_DIR = "/usr/local/yang-models"
+CHECKPOINTS_DIR = "/etc/sonic/checkpoints"
+CHECKPOINT_EXT = ".cp.json"
+
+class JsonChange:
+    # TODO: Implement JsonChange
+    pass
+
+class ConfigLock:
+    def acquire_lock(self):
+        # TODO: Implement ConfigLock
+        pass
+
+    def release_lock(self):
+        # TODO: Implement ConfigLock
+        pass
+
+class PatchOrderer:
+    def order(self, patch):
+        # TODO: Implement patch orderer
+        pass
+
+class ChangeApplier:
+    def apply(self, change):
+        # TODO: Implement change applier
+        pass
+
+class ConfigWrapper:
+    def __init__(self, default_config_db_connector = None, yang_dir = YANG_DIR):
+        self.default_config_db_connector = default_config_db_connector
+        self.yang_dir = YANG_DIR
+
+    def get_config_db_as_json(self):
+        config_db = self.__create_and_connect_config_db()
+        data = dict()
+        deep_update(data, FormatConverter.db_to_output(config_db.get_config()))
+        return FormatConverter.to_serialized(data)
+
+    def get_sonic_yang_as_json(self):
+        config_db_json = self.get_config_db_as_json()
+        return self.convert_config_db_to_sonic_yang(config_db_json)
+
+    def convert_config_db_to_sonic_yang(self, config_db_as_json):
+        sy = sonic_yang.SonicYang(self.yang_dir)
+        sy.loadYangModel()
+
+        # Crop config_db tables that do not have sonic yang models
+        cropped_config_db_as_json = self.crop_tables_without_yang(config_db_as_json)
+
+        sonic_yang_as_json = dict()
+
+        sy._xlateConfigDBtoYang(cropped_config_db_as_json, sonic_yang_as_json)
+
+        return sonic_yang_as_json
+
+    def convert_sonic_yang_to_config_db(self, sonic_yang_as_json):
+        sy = sonic_yang.SonicYang(self.yang_dir)
+        sy.loadYangModel()
+
+        # replace container of the format 'module:table' with just 'table'
+        new_sonic_yang_json = {}
+        for module_top in sonic_yang_as_json.keys():
+            new_sonic_yang_json[module_top] = {}
+            for container in sonic_yang_as_json[module_top].keys():
+                table = container.split(':')[1]
```

Comment:
>container [](http://example.com/codeflow?start=24&length=9)

`container` is from function input. So it may not contain a `:`. Then this line will throw.